src/core/libmaven/PeakGroup.cpp

```diff
@@ -1006,6 +1034,35 @@ string PeakGroup::tableName() const
 void PeakGroup::setTableName(string tableName)
 {
     _tableName = tableName;
-    for (auto child : children)
+    for (auto child : _childIsotopes)
+        child->setTableName(tableName);
+    for (auto child : _childAdducts)
```

Comment:
I think I saw a check for tableName in barplot somewhere. Can you check and update if that's the case?

Reply:
This is safe. The one in barplot is meant to check if the _parent_ peak-group is from a peak table or not. If so, then just display the already known child isotopes instead of finding and populating the barplot-specific isotopes (which can be using tracers). Barplot-specific isotopic groups are never presented to the user in any other way.